### PR TITLE
CI: automatically add a warning comment to PRs with backport labels

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
Core contributors are not all aligned and informed on what is acceptable to backport to stable and beta branches

#### Summary of Changes
Add a mergify action that will add a comment whenever a PR is labeled a `v1.16` or `v1.17` label. This extra comment will remind contributors and reviewers what the guidelines for backports are.

Fixes #

### CODE REVIEW

**Review by willhickey — APPROVED** [review]
Thanks!

File: .mergify.yml [hunk 1]
```diff
@@ -98,6 +98,17 @@ pull_request_rules:
...
+    conditions:
+      - label=v1.16
+    actions:
+      comment:
+        message: >
+          Backports to the stable branch are to be avoided unless absolutely
+          necessary for fixing bugs, security issues, and perf regressions.
+          Backported changes should be reduced to the smallest viable change,
+          refactoring should be done in a separate change that is not
+          backported.
```
Comment (t-nelson) [inline .mergify.yml:111]:
```suggestion
          Change intended for backport should be structured such that a
          minimum effective diff can be committed separately from any
          refactoring, plumbing, cleanup, etc that are not strictly
          necessary to achieve the goal. Any of the latter should go only
          into master and ride the normal stabilization schedule.
```

File: .mergify.yml [hunk 2]
```diff
@@ -122,6 +133,18 @@ pull_request_rules:
...
+      - label=v1.17
+    actions:
+      comment:
+        message: >
+          Backports to the beta branch are to be avoided unless absolutely
+          necessary for fixing bugs, security issues, and perf regressions.
+          Backported changes should be reduced to the smallest viable change,
+          refactoring should be done in a separate change that is not
+          backported. Exceptions include CI/metrics changes, CLI improvements,
+          and documentation updates.
```
Comment (t-nelson) [inline .mergify.yml:147]:
```suggestion
          Change intended for backport should be structured such that a
          minimum effective diff can be committed separately from any
          refactoring, plumbing, cleanup, etc that are not strictly
          necessary to achieve the goal. Any of the latter should go only
          into master and ride the normal stabilization schedule. Exceptions
          include CI/metrics changes, CLI improvements and documentation
          updates on a case by case basis.
```

**Review by t-nelson — APPROVED** [review]

**Comment by jstarry** [comment]
Hope it works

**Pull request merged**